report-ng/app/src/components/timings/session-timings/session-timings.ts

```diff
@@ -0,0 +1,289 @@
+import {AbstractViewModel} from "../../abstract-view-model";
+import {ECharts, EChartsOption} from 'echarts';
+import "./session-timings.scss";
+import {MethodDetails, StatisticsGenerator} from "services/statistics-generator";
+import {StatusConverter} from "../../../services/status-converter";
+import {MethodType, MetricType} from "../../../services/report-model/framework_pb";
+import {ExecutionStatistics} from "../../../services/statistic-models";
+import {
+    IntlDateFormatValueConverter
+} from "t-systems-aurelia-components/src/value-converters/intl-date-format-value-converter";
+
+@autoinject()
+export class SessionTimings extends AbstractViewModel {
+    private static readonly SESSION_COLOR = '#6897EA';
+    private static readonly BASEURL_COLOR = '#75C6CB';
+    private _chart: ECharts;
+    private _option: EChartsOption;
+    private _executionStatistics: ExecutionStatistics;
+    private _methodDetails: MethodDetails[];
+    private _testDuration: Duration | null = null;
+    private _dots: IDots[];
+
+    constructor(
+        private readonly _statusConverter: StatusConverter,
+        private _statisticsGenerator: StatisticsGenerator,
+        private readonly _dateFormatter: IntlDateFormatValueConverter,
+    ) {
+        super();
+        this._methodDetails = [];
+        this._dots = [];
+    }
+
+    activate(params: any, routeConfig: RouteConfig, navInstruction: NavigationInstruction) {
+        super.activate(params, routeConfig, navInstruction);
+
+        this._statisticsGenerator.getExecutionStatistics().then(executionStatistics => {
+            this._executionStatistics = executionStatistics;
+
+            this._testDuration = {
+                startTime: this._executionStatistics.executionAggregate.executionContext.contextValues.startTime,
+                endTime: this._executionStatistics.executionAggregate.executionContext.contextValues.endTime,
+            }
+
+            executionStatistics.classStatistics.forEach(classStatistic => {
+                const filteredMethodDetails = classStatistic.methodContexts
+                    .filter(methodContext => methodContext.methodType == MethodType.TEST_METHOD)
+                    .map(methodContext => new MethodDetails(methodContext, classStatistic));
+                this._methodDetails.push(...filteredMethodDetails);
+            })
+        });
+
+        const sessionInformationArray = [];
+
+        this._statisticsGenerator.getSessionMetrics().then(sessionMetrics => {
+            sessionMetrics.forEach(metric => {
+                const sessionData = metric.metricsValues.find(value => value.metricType === MetricType.SESSION_LOAD);
+                const baseurlData = metric.metricsValues.find(value => value.metricType === MetricType.BASEURL_LOAD);
+
+                if (!Object.prototype.hasOwnProperty.call(baseurlData, 'endTimestamp')){ // if there is no endTimestamp the related metric will be skipped
```

Comment:
This line causes an error in the browser console although this is the recommended way to check if properties exist. I already tried object.hasOwnProperty('property') and the "in" Operator, but they all throw an error if endTimestamp does not exist. I also read that Object.hasOwn() is recommended but this does not seem to work at all (maybe a version issue?)
Does anyone have any ideas how to solve this?

Reply:
```typescript
if (!metric.metricsValues[0]?.endTimestamp?) {
...
}
```

Reply:
With the `?` at the end of endTimestamp IntelliJ says it is expecting an expression. 
This seems to be working fine:  
`if (!(baseurlData?.endTimestamp > 0)){ 
                    return;
                }`
but SonarLint suggests writing it like that because otherwise this code produces a minor code smell:  
`if (baseurlData?.endTimestamp <= 0){ 
                    return;
                }`, which causes the same error message in the browser console as before. (Cannot read properties of undefined)
It does not make sense to me though, since it should be the same result. 
Should I just use the version that is working without the error message and ignore the SonarLint issue?

Reply:
@martingrossmann please respond.

Reply:
Please change it to `if (!(metric.metricsValues[0]?.endTimestamp > 0)) {`, it seems ok for me. 
Please do not leave the TS way.